lib/internal/util/comparisons.js

```diff
@@ -502,7 +502,7 @@ function setEquiv(a, b, strict, memo) {
     for (const val of b) {
       // Primitive values have already been handled above.
       if (typeof val === 'object' && val !== null) {
-        if (!setHasEqualElement(set, val, strict, memo))
+        if (!a.has(val) && !setHasEqualElement(set, val, strict, memo))
           return false;
```

Comment:
To help visually make this a bit easier to parse, let's wrap the body in brackets

```suggestion
        if (!a.has(val) && !setHasEqualElement(set, val, strict, memo)) {
          return false;
        }
```